### PR TITLE
[bgp] Refactor test_bgp_allow_list

### PR DESCRIPTION
**Pull request by yaqiangz**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

#### What is the motivation for this PR?
There is something confusing in current test_bgp_allow_list test. This PR is to refactor this case. It help us to add bgp_allow_list test scenes more easily in the future

#### How did you do it?
1. Modify variable name 
2. Modify code structure

#### How did you verify/test it?
 Run test_test_bgp_allow_list in t1 testbed. 
```
bgp/test_bgp_allow_list.py::test_default_allow_list_preconfig[1] PASSED [ 25%] 
bgp/test_bgp_allow_list.py::test_allow_list[1-permit] PASSED [ 50%]
bgp/test_bgp_allow_list.py::test_allow_list[1-deny] PASSED [ 75%]
bgp/test_bgp_allow_list.py::test_default_allow_list_postconfig[1] PASSED [100%]
```

### CODE REVIEW

**Comment by Blueve** [comment]
@StormLiangMS could you help review this change as well?
This is a refactor, we can easily add more test cases and reuse the framework in future.

File: tests/bgp/test_bgp_allow_list.py [hunk 1]
```diff
@@ -23,33 +19,14 @@
...
-
-TEST_COMMUNITY = '1010:1010'
-DROP_COMMUNITY = ''
-DEPLOYMENT_ID = '0'
-DEFAULT_ACTION = ''
-
 ALLOW_LIST = {
     'BGP_ALLOWED_PREFIXES': {
-        'DEPLOYMENT_ID|{}|{}'.format(DEPLOYMENT_ID, TEST_COMMUNITY): {
+        'DEPLOYMENT_ID|0|{}'.format(TEST_COMMUNITY): {
```
Comment (StormLiangMS) [inline tests/bgp/test_bgp_allow_list.py:24]: why change to 0 from DEPLOYMENT_ID?

File: tests/common/helpers/constants.py [hunk 2]
```diff
@@ -5,3 +5,18 @@
 ASIC_PARAM_TYPE_ALL = 'num_asics'
 ASIC_PARAM_TYPE_FRONTEND = 'frontend_asics'
 ASICS_PRESENT = 'asics_present'
+
+# Describe upstream neighbor of dut in different topos
+UPSTREAM_NEIGHBOR_MAP = {
+    "t0": "t1",
+    "t1": "t2",
+    "m0": "m1",
+    "t2": "t3"
```
Comment (StormLiangMS) [inline tests/common/helpers/constants.py:14]: There is no t3, upstream for t2 is wan router.

**Review by StormLiangMS — APPROVED** [review]
approved with 2 questions.

**Pull request merged**